src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtilsUnitTest.java

```diff
@@ -33,37 +37,163 @@
         final ArrayList<Integer> baseArgs = new ArrayList<>();
         baseArgs.add(DataSourceUtils.MIN_MAJOR_VERSION_NUMBER);
         baseArgs.add(DataSourceUtils.MIN_MINOR_VERSION_NUMBER);
-        baseArgs.add(DataSourceUtils.MIN_YEAR_RELEASED);
-        baseArgs.add(DataSourceUtils.MIN_MONTH_RELEASED);
-        baseArgs.add(DataSourceUtils.MIN_DAY_RELEASED);
+        baseArgs.add(DataSourceUtils.MIN_DATE.get(Calendar.YEAR));
+        baseArgs.add(DataSourceUtils.MIN_DATE.get(Calendar.MONTH));
+        baseArgs.add(DataSourceUtils.MIN_DATE.get(Calendar.DAY_OF_MONTH));
+
+        final ArrayList<Integer> goodRange = new ArrayList<>();
+        goodRange.add(DataSourceUtils.MAX_MAJOR_VERSION_NUMBER - DataSourceUtils.MIN_MAJOR_VERSION_NUMBER);
+        goodRange.add(DataSourceUtils.MAX_MINOR_VERSION_NUMBER - DataSourceUtils.MIN_MINOR_VERSION_NUMBER);
+        goodRange.add(
+                DataSourceUtils.MAX_DATE.get(Calendar.YEAR) - DataSourceUtils.MIN_DATE.get(Calendar.YEAR)
+        );
+        goodRange.add(
+                DataSourceUtils.MAX_DATE.get(Calendar.MONTH) - DataSourceUtils.MIN_DATE.get(Calendar.MONTH)
+        );
+        goodRange.add(
+                DataSourceUtils.MAX_DATE.get(Calendar.DAY_OF_MONTH) - DataSourceUtils.MIN_DATE.get(Calendar.DAY_OF_MONTH)
+        );
 
-        for ( int offset = -1 ; offset < 2; ++offset ) {
-            for ( int i = 0; i < baseArgs.size(); ++i ) {
+        final int MAX_OFFSET = 5;
 
-                final ArrayList<Object> argList = new ArrayList<>();
+        for ( int i = 0; i < baseArgs.size(); ++i ) {
+            for ( int offset = -MAX_OFFSET; offset < goodRange.get(i) + MAX_OFFSET; ++offset){
 
-                argList.addAll(baseArgs.subList(0, i));
+                final ArrayList<Object> argList = new ArrayList<>(baseArgs.subList(0, i));
                 argList.add(baseArgs.get(i) + offset);
 
-                if ( i < baseArgs.size() - 1) {
+                if ( i < baseArgs.size() - 1 ) {
                     argList.addAll(baseArgs.subList(i + 1, baseArgs.size()));
                 }
 
-                argList.add(offset >= 0);
+                try {
+                    final Calendar releaseDate = new GregorianCalendar(
+                            (int) argList.get(2), Utils.getCalendarMonth((int) argList.get(3)), (int) argList.get(4)
+                    );
+
+                    boolean passes = ((Integer) argList.get(0) >= DataSourceUtils.MIN_MAJOR_VERSION_NUMBER) && ((Integer) argList.get(0) <= DataSourceUtils.MAX_MAJOR_VERSION_NUMBER);
+                    passes = passes && ((Integer) argList.get(1) >= DataSourceUtils.MIN_MINOR_VERSION_NUMBER) && ((Integer) argList.get(1) <= DataSourceUtils.MAX_MINOR_VERSION_NUMBER);
```

Comment:
Again, what about the case mentioned above of 2.0 (major.minor) being between 1.5 and 3.0?

This `DataProvider` code is long and complicated and it's difficult for me to confirm that you've covered the important boundary cases. I'd recommend replacing it with a curated and easy-to-read but systematic set of test cases instead -- eg.,

```
* major < MIN_MAJOR (in which case minor shouldn't matter, so test with various values of minor)
* major = MIN_MAJOR (in which case minor does matter, so again test with various values of minor)
* major > MIN_MAJOR and < MAX_MAJOR (in which case minor shouldn't matter...)
* major = MAX_MAJOR (in which case minor does matter...)
* major > MAX_MAJOR (in which case minor shouldn't matter...)
```

Then have separate test cases for dates both within and outside the acceptable range, for cases where major and minor are in range.

Reply:
Sure.  I was trying to automate the tests based on the current acceptable range, then below I had specific test cases for the bounds checking.  I'll refactor as you suggested.

Reply:
The only issue here is that currently the minimum and maximum major version numbers are the same - `1`.  So some of the tests can't really be written (using `int` values).